tests/integration_tests.rs

```diff
@@ -24,6 +25,38 @@ use tokio::{select, try_join};
 use tonic_lnd::Client;
 
 const NONCE_BYTES: &[u8] = &[42u8; 16];
+
+// Helper function to get an invoice with a delay to ensure messenger initialization.
+// This is necessary because there is a race condition between the messenger initialization,
+// the internal lndk graph update, and calling get_invoice.
+// Before, get_invoice sometimes would fail because internally the graph has not been updated,
+// not finding the introduction node and failing when building the onion message.
+async fn get_invoice_with_delay(
+    handler: &Arc<OfferHandler>,
+    offer: &lightning::offers::offer::Offer,
+    amount: u64,
+    network: Network,
+    client: Client,
+    destination: Destination,
+    delay: Duration,
+) -> Result<(lightning::offers::invoice::Bolt12Invoice, u64, PaymentId), lndk::offers::OfferError> {
+    // Wait before calling get_invoice to allow messenger to initialize
+    tokio::time::sleep(delay).await;
```

Comment:
It'd be better to do a retry loop here with a timeout, if possible. Adding just a sleep can lead to flaky integration tests in the CI.